tools/cockpit.spec

```diff
@@ -481,22 +506,44 @@ authentication via sssd/FreeIPA.
 %attr(4750, root, cockpit-wsinstance) %{_libexecdir}/cockpit-session
 %{_datadir}/cockpit/branding
 
+# selinux
+%if 0%{?with_selinux}
+    %{_datadir}/selinux/packages/%{selinuxtype}/%{name}.pp.bz2
+    %{_mandir}/man8/%{name}_session_selinux.8.*
+    %{_mandir}/man8/%{name}_ws_selinux.8.*
+    %ghost %{_sharedstatedir}/selinux/%{selinuxtype}/active/modules/200/%{name}
+%endif
+
 %pre ws
 getent group cockpit-ws >/dev/null || groupadd -r cockpit-ws
 getent passwd cockpit-ws >/dev/null || useradd -r -g cockpit-ws -d /nonexisting -s /sbin/nologin -c "User for cockpit web service" cockpit-ws
 getent group cockpit-wsinstance >/dev/null || groupadd -r cockpit-wsinstance
 getent passwd cockpit-wsinstance >/dev/null || useradd -r -g cockpit-wsinstance -d /nonexisting -s /sbin/nologin -c "User for cockpit-ws instances" cockpit-wsinstance
+# selinux
```

Comment:
So these three script should run only when cockpit-ws actually ships those files (see `with_selinux` macro). It will fail especially in `%post` (`%pre` and `%postrun` will be just pointless operation (I think)).
Is there a good way of doing this? I think I cannot just use `with_selinux` macro as this happens in a different time. I could check if `%{_datadir}/selinux/packages/%{selinuxtype}/%{name}.pp.bz2` is present. Is there a better way of doing this?

Reply:
Why wouldn't wrapping this into `with_selinux` work? That's processed very early in the spec file, and is a purely textual operation. The post-processed `%post` textual contents would then just get copied verbatim. Did you try it? If it doesn't work, then the runtime `test -e` seems ok, but it'd IMHO be more elegant at rpm build time.